README.md

```diff
@@ -1,11 +1,188 @@
+
+![Bowling Scorecard](images/bowling.png)
+
+I enjoyed this challenge a lot more than last week's Ruby bowling scorecard challenge, mainly because I spent much more time planning, especially the scoring logic. The coding was therefore much more straightforward, even though it was in a relatively new language for me! I have included my
+[planning - including user stories - below](#planning).
+
+All of the classes are comprehensively tested with Jasmine except for `scoreprinter.js`. I was running low on time and wanted to get my scorecard displayed, so I created a basic 'stringifier' without test-driving it.
+
+Future possibilities:
+
+- messages to user displayed in browser
+- testing and refactoring for ScorePrinter class
+- display X for strike and / for spare
+- reset button
+- multiplayer function
+- lively CSS
+- more effective encapsulation?
+
+## Installation instructions
+
+```
+git clone https://github.com/edpackard/bowling-challenge.git
+```
+
+## How to use this program
+
+- Navigate to the `bowling-challenge` folder
+- Copy the path to `index.html` and paste it into your browser.
+- Click the buttons to enter each bowl score - the scorecard will update automatically.
+- You cannot enter an invalid bowl. At the moment, the error message is only displayed in the console.
+- The scorecard will no longer update when the game is complete. At the moment, the game complete message is only displayed in the console.
+- To reset the game, refresh your browser: there is currently no built-in reset function.
+
+## How to run the tests
+
+- Navigate to the `bowling-challenge` folder.
+- Copy the path to `SpecRunner.html` and paste it into your browser.
+- Look at the tests! The feature tests do not have expectations, they are there to simulate game inputs.
+
+## Planning
+
+**Frame Class Logic**
+
+```
+add(pins)
+---------
+
+Is frame full?
+    YES: Endpoint.
+    NO: Is pins an integer between 0 and number of remaining pins?
+        YES: Is it last frame?
+
+            YES: Is roll_1 null?
+                YES: Add pins to roll_1. Endpoint.
+                NO: Is roll_2 null?
+                    YES: Do roll_1 + pins equal 10 or more?
+                        YES: Add pins to roll_2, reset remaining pins. Endpoint.
+                        NO: Add pins to roll_2 and frame = full. Endpoint.
+                    NO: Add pins to roll_3 and frame = full. Endpoint.
+
+            NO: Is roll_1 null?
+                YES: Is pins value 10?
+                    YES: Add pins to roll_1 and set roll_2 to zero and frame = full. Endpoint.
+                    NO: Add pins to roll_1. Endpoint.
+                NO: Add pins to roll_2 and frame = full. Endpoint.
+        NO: Invalid roll. Endpoint.
+```
```

Comment:
Thank you for this really useful comment (I am struggling to find a good balance with my planning)  - these logic branches did kind of tie me to a particular way of dealing with the logic: the week before, with the Ruby version, I did try it with TDD and it was even more unwieldy than this (especially the scoring)! I You are absolutely right in that I probably need to step back and think of a different way of dealing with scoring. Some of your advice below will help with this I think.

Reply:
Copied from slack:

I think this comment is on point. SRP can be misunderstood and it can be hard to see how to separate classes, but separating calculations and the data that it uses to calculate is typically a bad move in OOP. In fact, you should think of a class as an entity that stores data, and present an interface to the world about this data, including calculations based on this data (which is a way to "see" the data in some way), and ways to change this data. If you want to encapsulate the rolls, it makes sense to me that the only way into your class is when you enter the roll, and the only ways of data out of the class are retrieving data about the score (total, partial, by frame, etc).
In that way, the frame class can also help by calculating score for an individual frame.